commands/deployments/dep_events.go

```diff
@@ -128,38 +128,86 @@ func StreamsEvents(client *httputil.YorcClient, deploymentID string, colorize, f
-				fmt.Printf("%s:\t Deployment: %s\t Node: %s\t Instance: %s\t State: %s\n", ts, event.DeploymentID, event.Node, event.Instance, event.Status)
-			case events.DeploymentStatusChangeType:
-				fmt.Printf("%s:\t Deployment: %s\t Deployment Status: %s\n", ts, event.DeploymentID, event.Status)
-			case events.CustomCommandStatusChangeType:
-				fmt.Printf("%s:\t Deployment: %s\t Task %q (custom command)\t Status: %s\n", ts, event.DeploymentID, event.TaskID, event.Status)
-			case events.ScalingStatusChangeType:
-				fmt.Printf("%s:\t Deployment: %s\t Task %q (scaling)\t Status: %s\n", ts, event.DeploymentID, event.TaskID, event.Status)
-			case events.WorkflowStatusChangeType:
-				fmt.Printf("%s:\t Deployment: %s\t Task %q (workflow)\t Status: %s\n", ts, event.DeploymentID, event.TaskID, event.Status)
-			}
-
+			fmt.Printf("%s\n", formatEvent(event, colorize))
 		}
-
 		response.Body.Close()
-
 		if stop {
 			return
 		}
 	}
 }
+
+func formatEvent(event json.RawMessage, colorize bool) string {
+	//fmt.Printf("%q", event)
+	var ret string
+	var data map[string]string
+	err := json.Unmarshal(event, &data)
+	if err != nil {
+		if colorize {
+			ret += fmt.Sprintf("%s: ", color.MagentaString("Warning"))
+		} else {
+			ret += fmt.Sprintf("Warning: ")
+		}
+		ret += fmt.Sprintf("Failed to unmarshal json:%q", event)
+		return ret
+	}
+
+	evType, is := data[events.EType.String()]
+	if !is {
+		if colorize {
+			ret += fmt.Sprintf("%s: ", color.MagentaString("Warning"))
+		} else {
+			ret += fmt.Sprintf("Warning: ")
+		}
+		ret += fmt.Sprintf("Misssing event type in json event:%q", event)
+		return ret
+	}
+
+	ts := data[events.ETimestamp.String()]
+	if colorize {
+		ts = color.CyanString("%s", ts)
+	}
+
+	statusChange, err := events.ParseStatusChangeType(evType)
+	if err != nil {
+		if colorize {
+			ret += fmt.Sprintf("%s: ", color.MagentaString("Warning"))
+		} else {
+			ret += fmt.Sprintf("Warning: ")
+		}
+		ret += fmt.Sprintf("Unknown event type: %q\n", evType)
+		return ret
+	}
+
+	switch statusChange {
+	case events.StatusChangeTypeInstance:
+		ret = fmt.Sprintf("%s:\t Deployment: %s\t Node: %s\t Instance: %s\t Status: %s\n", ts, data[events.EDeploymentID.String()], data[events.ENodeID.String()], data[events.EInstanceID.String()], data[events.EStatus.String()])
+	case events.StatusChangeTypeDeployment:
+		ret = fmt.Sprintf("%s:\t Deployment: %s\t Deployment Status: %s\n", ts, data[events.EDeploymentID.String()], data[events.EStatus.String()])
+	case events.StatusChangeTypeCustomCommand:
+		ret = fmt.Sprintf("%s:\t Deployment: %s\t Task %q (custom command)\t Status: %s\n", ts, data[events.EDeploymentID.String()], data[events.ETaskID.String()], data[events.EStatus.String()])
+	case events.StatusChangeTypeScaling:
+		ret = fmt.Sprintf("%s:\t Deployment: %s\t Task %q (scaling)\t Status: %s\n", ts, data[events.EDeploymentID.String()], data[events.ETaskID.String()], data[events.EStatus.String()])
+	case events.StatusChangeTypeWorkflow:
```

Comment:
It seems that Alien4Cloud is expecting a non-empty execution ID for a workflowSucceded event (or more generally any event related to a worflow finish).
I had this issue in Alien4Cloud at https://github.com/alien4cloud/alien4cloud/blob/develop/alien4cloud-rest-api/src/main/java/alien4cloud/rest/deployment/WorkflowEventHandler.java#L226 when it tries to find a workflow event by its execution ID :
```
2018-12-10 18:40:20 ERROR ESIndexMapper:135 - Null or empty Id is not allowed for operation <findById>.
2018-12-10 18:40:20 ERROR PaaSProviderPollingMonitor:167 - Failed to dispatch event PaaSWorkflowSucceededEvent() to listener alien4cloud.rest.deployment.WorkflowEventHandler@3e28af44 retry 3 on 3.
alien4cloud.exception.IndexingServiceException: Null or empty Id is not allowed for operation <findById>.
        at alien4cloud.dao.ESGenericIdDAO.assertIdNotNullFor(ESGenericIdDAO.java:136) ~[alien4cloud-dao-2.1.0-SM7.jar!/:?]
        at alien4cloud.dao.ESGenericIdDAO.findById(ESGenericIdDAO.java:84) ~[alien4cloud-dao-2.1.0-SM7.jar!/:?]
        at alien4cloud.dao.ESGenericIdDAO$$FastClassBySpringCGLIB$$7977fa30.invoke(<generated>) ~[alien4cloud-dao-2.1.0-SM7.jar!/:?]
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) ~[spring-core-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:720) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at alien4cloud.dao.ESExceptionHandlingAspect.handleException(ESExceptionHandlingAspect.java:26) ~[alien4cloud-dao-2.1.0-SM7.jar!/:?]
        at sun.reflect.GeneratedMethodAccessor133.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_162]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_162]
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:629) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:618) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:168) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:655) ~[spring-aop-4.3.2.RELEASE.jar!/:4.3.2.RELEASE]
        at alien4cloud.dao.ElasticSearchDAO$$EnhancerBySpringCGLIB$$213116ca.findById(<generated>) ~[alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.rest.deployment.WorkflowEventHandler.updateExecution(WorkflowEventHandler.java:226) ~[alien4cloud-rest-api-2.1.0-SM7.jar!/:?]
        at alien4cloud.rest.deployment.WorkflowEventHandler.send(WorkflowEventHandler.java:80) ~[alien4cloud-rest-api-2.1.0-SM7.jar!/:?]
        at alien4cloud.rest.deployment.WorkflowEventHandler.eventHappened(WorkflowEventHandler.java:321) ~[alien4cloud-rest-api-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor.dispatchEvent(PaaSProviderPollingMonitor.java:164) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor.dispatchEvent(PaaSProviderPollingMonitor.java:171) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor.dispatchEvent(PaaSProviderPollingMonitor.java:171) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor.dispatchEvent(PaaSProviderPollingMonitor.java:171) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor.dispatchEvent(PaaSProviderPollingMonitor.java:150) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor.access$500(PaaSProviderPollingMonitor.java:28) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor$PaaSEventsCallback.onSuccess(PaaSProviderPollingMonitor.java:119) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor$PaaSEventsCallback.onSuccess(PaaSProviderPollingMonitor.java:91) [alien4cloud-core-2.1.0-SM7.jar!/:?]
        at org.ystia.yorc.alien4cloud.plugin.YorcPaaSProvider.getEventsSince(YorcPaaSProvider.java:357) [0e78b476-ad14-47fe-904a-d6d1ef1ec032/:?]
        at alien4cloud.paas.PaaSProviderPollingMonitor.run(PaaSProviderPollingMonitor.java:190) [alien4cloud-core-2.1.0-SM7.jar!/:?]
```

Reply:
so the issue should be in Alien4Cloud using the executionID instead of the wokflowID to identify workflow events

Reply:
IMO it's correct to expect an existing executionID for a workflow related event. That's why the field is mandatory here: https://github.com/ystia/yorc/blob/feature/gh-148-events/events/structs.go#L184

Reply:
Maybe something is wrong in other part ?

Reply:
My comment is not relevant as I was testing a version that I modified to invert task ID and execution ID, being not aware of the naming issue between Yorc and Alien4Cloud.
I was trying so see if this change would fix an issue we see with the current version in the PR where in Alien4Cloud, for a given deployment, going to the Executions tab, selecting a worflow, trying to access logs of a delegate operation, no log is shown anymore.